### PR TITLE
[oneDNN] Cache oneDNN stream not to recreate in each oneDNN op 

### PR DESCRIPTION
**Pull request by jczaja**
### PR types
Performance optimization

### PR changes
OPs

### Describe
created on dnnl::stream involves small memory allocation. stream can be created once and then reused so this PR makes stream to be created one per thread and store in per thread object.

### CODE REVIEW

File: paddle/fluid/platform/device_context.h [hunk 1]
```diff
@@ -525,6 +525,9 @@ class MKLDNNDeviceContextThreadLocals {
     // Recently registered data_format. This is needed to
     // know for converting MKL-DNN Tensor to non MKL-DNN
     paddle::framework::DataLayout cur_paddle_data_layout;
+    // MKL-DNN stream used for execution of primitives (per-thread)
+    mkldnn::engine aengine;
+    mkldnn::stream astream;
```
Comment (arogowie-intel) [inline paddle/fluid/platform/device_context.h]: The names of these variables don't stick to current naming convention in this struct. I'd suggest to remove the `a` prefix.

File: paddle/fluid/operators/fused/mkldnn/fusion_gru_mkldnn_op.cc [hunk 2]
```diff
@@ -246,7 +246,7 @@ class GRUMKLDNNHandler : public platform::MKLDNNHandlerT<T, dnnl::gru_forward> {
       memory_p = std::make_shared<dnnl::memory>(this->fwd_pd_->src_iter_desc(),
                                                 this->engine_);
 
-      dnnl::stream astream(this->engine_);
+      auto astream = paddle::platform::MKLDNNDeviceContext::tls().get_stream();
```
Comment (arogowie-intel) [inline paddle/fluid/operators/fused/mkldnn/fusion_gru_mkldnn_op.cc]: Please consider this change to make sure no copy is done.
```suggestion
      const auto& astream = paddle::platform::MKLDNNDeviceContext::tls().get_stream();
```
The `execute` function accepts `astream` arg as a [const ref already](https://spec.oneapi.com/versions/latest/elements/oneDNN/source/primitives/reorder.html#_CPPv4NK4dnnl7reorder7executeERK6streamR6memoryR6memory).

**Review by arogowie-intel — COMMENTED** [review]
Please use `auto&` to make sure that you're not doing copy while reaching for stream object.

File: paddle/fluid/platform/device_context.h (hunk 1 above)
Comment (jczaja) [reply paddle/fluid/platform/device_context.h]: ok

File: paddle/fluid/operators/fused/mkldnn/fusion_gru_mkldnn_op.cc (hunk 2 above)
Comment (jczaja) [reply paddle/fluid/operators/fused/mkldnn/fusion_gru_mkldnn_op.cc]: ok . done

**Comment by jczaja** [comment]
@arogowie-intel I have added some additional change (last commit) to fix a problem exposed by test-analyzer-detect. If you want more details then ask me.

**Review by jakpiase — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@luotao1 Could you please start your review? CI will not pass without approval. This PR is changing way oneDNN stream was created , so due to API change I had do it all at once in all locations. So there many files modified, but most of them it is single line of change.

**Review by arogowie-intel — APPROVED** [review]

**Review by luotao1 — APPROVED** [review]

**Pull request merged**